cpp/src/arrow/meson.build

```diff
@@ -356,38 +356,82 @@ install_headers(
 )
 
 if needs_tests
+    boost_dep = dependency('boost', include_type: 'system', required: false)
+
+    asio_dep = dependency(
+        'boost',
+        include_type: 'system',
+        modules: 'asio',
+        required: false,
+    )
+
     filesystem_dep = dependency(
         'boost',
-        modules: ['filesystem'],
+        include_type: 'system',
+        modules: 'filesystem',
         required: false,
     )
-    if not filesystem_dep.found()
+
+    process_dep = dependency(
+        'boost',
+        include_type: 'system',
+        modules: 'process',
+        required: false,
+    )
+
+    if not (boost_dep.found()
+and asio_dep.found()
+and filesystem_dep.found()
+and process_dep.found()
+)
```

Comment:
Oh... This is a weird format by `meson format`...

Reply:
I agree. I've opened https://github.com/mesonbuild/meson/issues/14364 upstream for now. We could also refactor the code if there isn't much appetite upstream to change